File: app/src/main/java/com/quintype/oxygen/models/storypresenter/ElementViewType.java
package com.quintype.oxygen.models.storypresenter;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.SOURCE)
public @interface ElementViewType {
    int TEXT = 1;
    int IMAGE = 2;
    int YOUTUBE = 3;
    int SOUND_CLOUD = 4;
    int TITLE = 5;
    int SUMMARY = 6;
    int QUOTE = 7;
    int AUDIO = 8;
    int GIF = 9;
    int BLURB = 16;
    int BIG_FACT = 17;
    int SLIDESHOW = 18;
    int GALLERY = 19;
    int BLOCK_QUOTE = 20;
    int MEDIA = 21;
    int VIDEO = 22;
    int JS_EMBED = 23;
    int TWEET = 24;
    int QUESTION_ANSWER = 25;
    int BIT_GRAVITY_VIDEO = 32;
    int BRIGHTCOVE_VIDEO = 33;
    int POLLTYPE = 34;
    int ALSO_READ = 35;
    int QUESTION = 36;
    int ANSWER = 37;
    int TIME_STAMP = 38;
    int SORT_CARDS = 39;
    int NESTED_STORY = 40;
    int TABLE = 10;
    int UNKNOWN = 48;

    int STORY_HERO_AUTHOR = 51;
    int STORY_TAG = 52;
    int ME_TYPE = 53;
    int STORY_FEEDBACK = 54;
    int ATTACHMENT = 55;
    int REFERENCE = 56;

    int ICON_WITH_TITLE = 57;
    int SOCIAL_SHARE_WIDGET = 58;
    int CONTRIBUTORS_WIDGET = 59;
    int ME_TYPE_REACTION = 60;
    int ME_TYPE_CLAP = 61;
    int ADS = 62;
    int CARD_SHARE_WIDGET = 63;

    class Process {
        public Process() {
        }

        static int getType(int type) {
            return type;
        }
    }
}

File: app/src/main/java/com/quintype/oxygen/models/storypresenter/ElementViewType.kt
package com.quintype.oxygen.models.storypresenter

import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Retention(RetentionPolicy.SOURCE)
annotation class ElementViewType {

    class Process {
        companion object {
            internal fun getType(type: Int): Int {
                return type
            }
        }
    }

    companion object {
        val TEXT = 1
        val IMAGE = 2
        val YOUTUBE = 3
        val SOUND_CLOUD = 4
        val TITLE = 5
        val SUMMARY = 6
        val QUOTE = 7
        val AUDIO = 8
        val GIF = 9
        val BLURB = 16
        val BIG_FACT = 17
        val SLIDESHOW = 18
        val GALLERY = 19
        val BLOCK_QUOTE = 20
        val MEDIA = 21
        val VIDEO = 22
        val JS_EMBED = 23
        val TWEET = 24
        val QUESTION_ANSWER = 25
        val BIT_GRAVITY_VIDEO = 32
        val BRIGHTCOVE_VIDEO = 33
        val POLLTYPE = 34
        val ALSO_READ = 35
        val QUESTION = 36
        val ANSWER = 37
        val TIME_STAMP = 38
        val SORT_CARDS = 39
        val NESTED_STORY = 40
        val TABLE = 10
        val UNKNOWN = 48

        val STORY_HEADER = 51
        val STORY_TAG = 52
        val ME_TYPE = 53
        val STORY_FEEDBACK = 54
        val ATTACHMENT = 55
        val REFERENCE = 56
        val ME_TYPE_REACTION = 60
        val ME_TYPE_CLAP = 61

        val ICON_WITH_TITLE = 57
        val SOCIAL_SHARE_WIDGET = 58
        val CONTRIBUTORS_WIDGET = 59
        val ADS = 60
        val CARD_SHARE_WIDGET = 61
    }
}